.github/workflows/lib-injection.yml

```diff
@@ -48,63 +48,117 @@ jobs:
           LINUX_PACKAGE=linux-musl-arm
+
+    - name: Publish multiarch images with all platforms
+      shell: bash
+      run: |
+        # Issue: The above 'docker buildx build' commands only create an image for one platform at a time because we have different build-args for each invocation.
+        # This results in the 'dd-lib-dotnet-init:<SHA>[-musl]' tag being overwritten by the latest build command, so the result is not a multiarch image
+
+        # Fix: The following lines create and publish a new manifest to overwrite the 'dd-lib-dotnet-init:<SHA>[-musl]' tag.
+        # In the new manifest we directly reference each platform-specific image, so the tag becomes a multiarch image.
+        
+        docker manifest create \
+          ghcr.io/datadog/dd-trace-dotnet/dd-lib-dotnet-init:${{ github.sha }} \
+          --amend "ghcr.io/datadog/dd-trace-dotnet/dd-lib-dotnet-init@${{steps.build-image-amd64.outputs.digest}}" \
+          --amend "ghcr.io/datadog/dd-trace-dotnet/dd-lib-dotnet-init@${{steps.build-image-arm64v8.outputs.digest}}"
+        docker manifest push ghcr.io/datadog/dd-trace-dotnet/dd-lib-dotnet-init:${{ github.sha }}
+
+        docker manifest create \
+          ghcr.io/datadog/dd-trace-dotnet/dd-lib-dotnet-init:${{ github.sha }}-musl \
+          --amend "ghcr.io/datadog/dd-trace-dotnet/dd-lib-dotnet-init@${{steps.build-image-musl-amd64.outputs.digest}}" \
+          --amend "ghcr.io/datadog/dd-trace-dotnet/dd-lib-dotnet-init@${{steps.build-image-musl-arm64v8.outputs.digest}}"
+        docker manifest push ghcr.io/datadog/dd-trace-dotnet/dd-lib-dotnet-init:${{ github.sha }}-musl
+
+  test:
+    needs:
+      - build-and-publish-init-image
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+    strategy:
+      matrix:
+        lib-injection-connection: ['network','uds']
+        lib-injection-use-admission-controller: ['', 'use-admission-controller']
+        weblog-variant: ['dd-lib-dotnet-init-test-app']
+        runtime: ['bullseye-slim','alpine']
+        include:
+          - runtime: 'bullseye-slim'
+            init-tag-suffix: ''
+          - runtime: 'alpine'
+            init-tag-suffix: '-musl'
+      fail-fast: false
+    env:
+      TEST_LIBRARY: dotnet
+      WEBLOG_VARIANT: ${{ matrix.weblog-variant }}
+      LIBRARY_INJECTION_CONNECTION: ${{ matrix.lib-injection-connection }}
+      LIBRARY_INJECTION_ADMISSION_CONTROLLER: ${{ matrix.lib-injection-use-admission-controller }}
+      DOCKER_REGISTRY_IMAGES_PATH: ghcr.io/datadog
+      DOCKER_IMAGE_TAG: ${{ github.sha }}${{ matrix.init-tag-suffix }}
+      DOCKER_IMAGE_WEBLOG_TAG: ${{ github.sha }}-${{ matrix.runtime }}
+      RUNTIME: ${{ matrix.runtime }}
+      BUILDX_PLATFORMS: linux/amd64
+    steps:
+      - name: lib-injection test runner
+        id: lib-injection-test-runner
+        uses: DataDog/system-tests/lib-injection/runner@69336117276b3ef31f2594d20cbd8cd8a7d26396 # zach/lib-injection-dotnet
```

Comment:
Looks like something that will break one day :)

Reply:
I had to set this to my dev branch to test the changes, but we can either keep this on a `main` SHA or always use the latest commit of the repo in general.

This is why the PR is `do-not-merge` 😄

Reply:
Since I've now merged the system-tests PR, I've updated this to reference the `main` branch 😄